src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -88,6 +100,46 @@ internal class RenameEndpoint : IRenameEndpoint
                 return null;
             }
 
+            var razorEdits = await TryGetRazorComponentRenameEditsAsync(request, requestDocumentSnapshot, codeDocument, cancellationToken).ConfigureAwait(false);
+
+            if (razorEdits is not null)
+            {
+                return razorEdits;
+            }
+
+            // If we're not doing single server rename, then we're done. C# and Html will be handled by the RenameHandler in the HtmlCSharp server.
+            if (!_languageServerFeatureOptions.SingleServerRenameSupport)
```

Comment:
Would it help to rename the old endpoint as `LegacyRenameEndpoint` and then re-purpose `RenameEndpoint` to always assume single server support is on? That way we could be consistent with the pre-existing "single completion" like model where we [register different services depending on the flag](https://github.com/dotnet/razor-tooling/blob/e95f3afeaca8c408744bd59e2a849fbbf63fd714/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs#L167). Ultimately that design ensures we don't accidentally break anything.

Reply:
Personally, I don't think this would help, no. It would mean copying the code that is here, or sharing it in some way, then copying all of the tests for this endpoint and duplicating them to call the new endpoint, and I don't see what that buys us.

For completion I think it makes sense, as there are more logic changes there, but in this case where the handler is doing the same thing in both scenarios, and then doing a little bit more in single server mode, I think its easier to read and less code to maintain to keep things together. To me it feels like _this_ approach will ensure we don't accidentally break anything.

I also like that I can set a breakpoint in this endpoint, and know it will be hit, because there is no "hidden" logic being done to tweak DI or capabilities based on a feature flag.


BUT I'm fine to change it if you think its better. Consistency is definitely important, but I also feel like this, and completion, are different beasts.

Reply:
> BUT I'm fine to change it if you think its better. Consistency is definitely important, but I also feel like this, and completion, are different beasts.

I'm totally cool rolling with it too. As long as we have confidence that we wont break pre-existing experiences, have a way to remotely turn things off and have an easy way to "remove" all the legacy code once we're on-by-default then my concerns are quelled 😄